src/main/java/com/conveyal/taui/controllers/BundleController.java

```diff
@@ -204,11 +219,59 @@ public static Bundle update (Request req, Response res) throws IOException {
         return Persistence.bundles.updateFromJSONRequest(req);
     }
 
-    public static Object getBundle (Request req, Response res) {
-        return Persistence.bundles.findByIdFromRequestIfPermitted(req);
+    public static Bundle getBundle (Request req, Response res) {
+        Bundle bundle = Persistence.bundles.findByIdFromRequestIfPermitted(req);
+
+        // Old bundles did not have these set.
+        try {
+            setBundleServiceDates(bundle);
+        } catch (Exception e) {
+            throw AnalysisServerException.unknown(e);
+        }
+
+        return bundle;
+    }
+
+    public static Collection<Bundle> getBundles (Request req, Response res) {
+        return Persistence.bundles.findPermittedForQuery(req);
+    }
+
+    /**
+     * Old bundles were created without computing the service start and end dates. This finds them and stores them.
+     * @param bundle
+     */
+    public static Bundle setBundleServiceDates (Bundle bundle) throws Exception {
+        if (bundle.status != Bundle.Status.DONE || (bundle.serviceStart != null && bundle.serviceEnd != null)) return bundle;
+
+        // Old bundles were created with computing the service start and end dates
+        bundle.serviceStart = LocalDate.MAX;
+        bundle.serviceEnd = LocalDate.MIN;
+
+        for (Bundle.FeedSummary summary : bundle.feeds) {
+            // Compute the feed start and end dates
+            if (summary.serviceStart == null || summary.serviceEnd == null) {
+                FeedSource fs = ApiMain.getFeedSource(Bundle.bundleScopeFeedId(summary.feedId, bundle._id));
+                // Set service start and end from the dates of service
+                List<LocalDate> datesOfService = fs.feed.getDatesOfService();
+                datesOfService.sort(Comparator.naturalOrder());
+                summary.serviceStart = datesOfService.get(0);
+                summary.serviceEnd = datesOfService.get(datesOfService.size() - 1);
+            }
+
+            if (summary.serviceStart.isBefore(bundle.serviceStart)) {
+                bundle.serviceStart = summary.serviceStart;
+            }
+            if (summary.serviceEnd.isAfter(bundle.serviceEnd)) {
+                bundle.serviceEnd = summary.serviceEnd;
+            }
+        }
+
+        // Automated change that could occur on a `get`, so don't update the nonce
+        return Persistence.bundles.modifiyWithoutUpdatingLock(bundle);
```

Comment:
This feels a bit like we're changing / making exceptions to our model for storing and updating just to avoid a database migration. That migration is then becoming a permanent part of the code base. Should we store these date ranges for all newly created bundles, and migrate the old ones? Do we have any other places that we migrate to a newer data model automatically on get requests?

Reply:
`modifyWithoutUpdatingLock` is something that should've existed already and in another place a method was used that still updated the lock even though the comments surrounding it showed intent not too.

If it was purely MongoDB data I would just run a migration, but this would require loading all of the individual feeds in which is not the smallest operation. I'm unsure how many active bundles we have in the system and how long that would take, so this seemed like a place that progressive updates would be acceptable.